prime-dotnet-webapi/Services/AgreementService.cs

```diff
@@ -66,12 +66,15 @@ public async Task<int> GetLatestAgreementVersionIdOfTypeAsync(AgreementType type
                 .FirstAsync();
         }
 
-        public async Task<SignedAgreementDocument> AddSignedAgreementDocumentAsync(int agreementId, Guid documentGuid)
+        public async Task<SignedAgreementDocument> AddSignedAgreementDocumentAsync(int agreementId, Guid documentGuid, string filename = "")
         {
-            var filename = await _documentClient.FinalizeUploadAsync(documentGuid, DestinationFolders.SignedAgreements);
-            if (string.IsNullOrWhiteSpace(filename))
+            if (filename == null)
```

Comment:
Should this comparison be with empty string rather than `null` ... I'm not sure ... related to this, when is `SubmissionsController.AcceptToa` called?

Reply:
good catch. it is fixed.